api/odl.py

```diff
@@ -890,3 +956,509 @@ def _url_for(self, *args, **kwargs):
         del kwargs["_external"]
         return "http://%s?%s" % ("/".join(args), "&".join(["%s=%s" % (key, val) for key, val in kwargs.items()]))
 
+
+class SharedODLAPI(BaseCirculationAPI):
+    """An API for circulation managers to use to connect to an ODL collection that's shared
+    by another circulation manager.
+    """
+    NAME = "Shared ODL For Consortia"
+    DESCRIPTION = _("Import books from an ODL collection that's hosted by another circulation manager in the consortium. If this circulation manager will be the main host for the collection, select %(odl_name)s instead.", odl_name=ODLWithConsolidatedCopiesAPI.NAME)
+
+    SETTINGS = [
+        {
+            "key": Collection.EXTERNAL_ACCOUNT_ID_KEY,
+            "label": _("Base URL"),
+            "description": _("The base URL for the collection on the other circulation manager."),
+        },
+        {
+            "key": Collection.DATA_SOURCE_NAME_SETTING,
+            "label": _("Data source name"),
+        },
+    ]
+
+    SUPPORTS_REGISTRATION = True
+
+    TIME_FORMAT = "%Y-%m-%dT%H:%M:%SZ"
+
+    CHECKOUT_URL = "%(base_url)s/%(identifier_type)s/%(identifier)s/borrow"
+    HOLD_URL = "%(base_url)s/%(identifier_type)s/%(identifier)s/place_hold"
+
+    def __init__(self, _db, collection):
+        if collection.protocol != self.NAME:
+            raise ValueError(
+                "Collection protocol is %s, but passed into ODLWithConsolidatedCopiesAPI!" %
+                collection.protocol
+            )
+        self.collection_id = collection.id
+        self.data_source_name = collection.external_integration.setting(Collection.DATA_SOURCE_NAME_SETTING).value
+        # Create the data source if it doesn't exist yet.
+        DataSource.lookup(_db, self.data_source_name, autocreate=True)
+
+        self.base_url = collection.external_account_id
+
+    def internal_format(self, delivery_mechanism):
+        """Each consolidated copy is only available in one format, so we don't need
+        a mapping to internal formats.
+        """
+        return delivery_mechanism
+
+    def collection(self, _db):
+        return get_one(_db, Collection, id=self.collection_id)
+
+    def _get(self, url, headers=None, patron=None, allowed_response_codes=None):
```

Comment:
There is a decent amount of code in this method and taking `HTTP.get_with_timeout` as an argument would make it easy to test.

Reply:
Or even better, making it a constructor argument so all the other methods can use it. But I haven't looked at your tests yet so don't know how you handle it now.